lib/arsi/persistence.rb

```diff
@@ -0,0 +1,26 @@
+module Arsi
+  module Persistence
+    def self.prepended(klass)
+      klass.singleton_class.prepend(ClassMethods)
+    end
+
+    module ClassMethods
+      def _update_record(*)
+        with_relation_in_connection { super }
+      end
+
+      def _delete_record(*)
+        with_relation_in_connection { super }
+      end
+
+      private
+
+      def with_relation_in_connection
+        connection.arsi_relation = self.unscoped
```

Comment:
The existing code expects an instance of `Relation`, not a subclass of `ActiveRecord::Base`. Therefore we somewhat redundantly generate a scope here.